third_party/CMakeLists.txt

```diff
@@ -308,8 +308,13 @@ if (NOT (DEFINED ASTC_ARCH))
     # `ld: object file .../libastcdec-avx2-static.a was built for different x86_64 sub-type (8) than link command line (3)`
     # so we fallback to native. If we're on an ARM based mac, there's no need for any SIMD extension, because they support
     # ASTC_LDR natively, so this library won't even be used.
+    # Always build for ARM when compiling or cross-compiling to iOS physical devices, otherwise fall back to native as above
     if (APPLE)
-        set(ASTC_ARCH NATIVE)
+		if(IOS AND ${CMAKE_OSX_SYSROOT} STREQUAL "iphoneos")
```

Comment:
Does this work for iPad?

Great catch by the way!

Reply:
@gpx1000 yes it should work for iPad as is (which is an iOS target from a developer perspective).  As far as I know, the valid Apple target platforms (cmake SysRoots) are:

![Screenshot 2025-03-11 at 5 46 38 PM](https://github.com/user-attachments/assets/92442305-7532-445f-95bf-84c561daf72c)

This project supports three of them: _macosx_, _iphoneos_, and _iphonesimulator_.  While I have not tested on a physical iPad (don't have one), I have tested using a virtual iPad using the _iphonesimulator_ target, and that works fine.  If you have a physical iPad, perhaps you could test and let me know if there are any issues.  Thx.

Reply:
Yep, that looks right, the CMake docs say that it could be appleTV, iOS, MacOS, WatchOS, or visionOS etc.  I'm okay with the changes here thanks!